### ISSUE TITLE
assert doesn't work

### ISSUE THREAD

**Issue by mrkam2** [clj-kondo/clj-kondo#2256]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2023.10.20

**problem**

When `assert` is used in the hook, linting fails with `error: No matching ctor found for class java.lang.AssertionError`.

**repro**

`a.clj`:
```clojure
(ns a)

(defmacro my-macro [& body]
  `(do ~@body))

(my-macro :valid)

(my-macro)

```

`.clj-kondo/hooks/a.clj`:
```clojure
(ns hooks.a)

(defmacro my-macro [& body]
  (assert (seq body) "Body can't be empty")
  `(do ~@body))
```

**config**

`.clj-kondo/config.edn`:
```clojure
{:hooks {:macroexpand {a/my-macro hooks.a/my-macro}}}
```

All files in zip: [assert.zip](https://github.com/clj-kondo/clj-kondo/files/13867248/assert.zip).

**expected behavior**

`assert` is failing linting with the specified message when the condition is violated and produces no message otherwise.

**Comment by borkdude** [comment]
- [ ] TODO, `assert` throws `Error` rather than an `Exception`, so we should catch `Throwable` rather than `Exception` when handling exceptions
